lib/Sema/TypeCheckExprObjC.cpp

```diff
@@ -215,8 +215,16 @@ Optional<Type> TypeChecker::checkObjCKeyPathExpr(DeclContext *dc,
                diag::expr_unsupported_objc_key_path_component,
                (unsigned)kind);
       continue;
-    case KeyPathExpr::Component::Kind::OptionalWrap:
     case KeyPathExpr::Component::Kind::Property:
+    case KeyPathExpr::Component::Kind::Type: {
+      //For code completion purposes, we only care about the last expression.
```

Comment:
It's not clear to me how this relates to code completion.

Reply:
I might have treated this in the wrong place, but the issue I was having was that this method seems to be called twice on code completion - first to resolve the properties and then again in order to return the property that's trying to be completed, as far as I could tell. I couldn't find another case where this happened, so I assumed that code completion was the only case where the resolved enum cases would trigger here. Is it the wrong place?

The main idea for this block is that if I had something like`#keyPath(MyType.myProperty.`, I had to ignore the type and return `myProperty` for code completion to work.